### PR TITLE
[Interop] Add circom circuits

### PR DESCRIPTION
**Pull request by biscuitdey**
# Description

Add the circom circuits for the interop demo. The circuit includes the non-inclusion merkle proof verifier and ecdsa signature verifier.

## Related Issue

#724 

## Motivation and Context

Interop demo circuits

## How Has This Been Tested

Tests will be added in future PRs

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Request to be added as a Code Owner/Maintainer

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I commit to abide by the Responsibilities of Code Owners/Maintainers.

### CODE REVIEW

File: examples/bpi-interop/circuits/circuit.circom [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+include "../../../../../../node_modules/circomlib/circuits/comparators.circom";
+include "./utils/ecdsaSignatureVerifier.circom";
+include "./utils/merkleProofVerifier.circom";
+include "./utils/arithmeticOperators.circom";
+
+template Circuit(nodes){
+
+	//Non-inclusion Merkle Proof inputs
+	signal input merkelizedInvoiceRoot;
```
Comment (Therecanbeonlyone1969) [inline examples/bpi-interop/circuits/circuit.circom:11]: should that not be `merkelizedBankStateObject`?

File: examples/bpi-interop/circuits/circuit.circom [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	signal input stateTreeRoot;
+	signal input stateTree[nodes];
+	signal input stateTreeLeafPosition[nodes];
```
Comment (Therecanbeonlyone1969) [inline examples/bpi-interop/circuits/circuit.circom:14]: Same comment as above. Reference to `BankStateObject`

File: examples/bpi-interop/circuits/circuit.circom [hunk 3]
```diff
@@ -0,0 +1,63 @@
+pragma circom 2.1.5;
+
+include "../../../../../../node_modules/circomlib/circuits/comparators.circom";
+include "./utils/ecdsaSignatureVerifier.circom";
+include "./utils/merkleProofVerifier.circom";
+include "./utils/arithmeticOperators.circom";
+
+template Circuit(nodes){
+
+	//Non-inclusion Merkle Proof inputs
```
Comment (Therecanbeonlyone1969) [inline examples/bpi-interop/circuits/circuit.circom:10]: And we need proof of inclusion, not non-inclusion.

File: examples/bpi-interop/circuits/circuit.circom [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+	signal input merkelizedInvoiceRoot;
+	signal input stateTreeRoot;
+	signal input stateTree[nodes];
+	signal input stateTreeLeafPosition[nodes];
+	
+	//Signature inputs
```
Comment (Therecanbeonlyone1969) [inline examples/bpi-interop/circuits/circuit.circom:16]: We need the verification of signatures of ACME and ACME Customer

**Review by Therecanbeonlyone1969 — COMMENTED** [review]
@biscuitdey Looks good ... left a couple of comments. The primary comment is that we need to verify the signature of the ACME customer over the bank data object and the signature of ACME over the bank data object signed by the ACME customer.

**Comment by ognjenkurtic** [comment]
Core-devsw 30.10.2023.
@biscuitdey will look into the open questions and address them so that we can merge this one.

**Comment by Ybittan** [comment]
Core dev 11/13 - Pending @biscuitdey to review comments

**Comment by biscuitdey** [comment]
This PR uses Ecdsa circuits. As we are moving to Eddsa signatures in the circom circuits, I am closing this PR. 
Will raise a new PR with Eddsa signatures (as and when they are finalised).

**Pull request closed without merging**